com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/DebugAdapter.java

```diff
@@ -125,6 +126,7 @@ private void initialize() {
         registerHandlerForDebug(new SetDataBreakpointsRequestHandler());
         registerHandlerForDebug(new InlineValuesRequestHandler());
         registerHandlerForDebug(new RefreshVariablesHandler());
+        registerHandlerForDebug(new ProcessIdHandler());
 
         // NO_DEBUG mode only
         registerHandlerForNoDebug(new DisconnectRequestWithoutDebuggingHandler());
```

Comment:
When I debug Boot App via "Debug" action i get this back:
```
{"processId":0,"shellProcessId":78134}
```
which is great.

However, when i run the app via the "Run" action i get an error back:
```
Error: {"name":"Error","message":"custom request failed"}
```

Is it possible to make it work with "Run" action? 

I thought the change for this would be instead of `registerHandlerForDebug(...)` call `registerHandler(...)` but this didn't work for me and i hope also because i messed up the build...

What do you think?

Reply:
yes, we can support it in "Run" as well.

Reply:
I have added 'processId' request in Run action as well.

If the Run action is running with internalConsole, it works fine. 

However, if the Run action is triggered on terminal shell, it may not work. In this case, Java debugger will terminate the DAP debug session early once the RunInTerminal request is responded. You may not have opportunity to call custom request to get processId.

We have a TODO task to not terminate DAP session when triggering a Run with integratedTerminal, which requires more work and may not be done right away.